v0/destinations/marketo_bulk_upload/fileUpload.js

```diff
@@ -15,11 +15,44 @@ const {
   removeUndefinedAndNullValues,
   isDefinedAndNotNullAndNotEmpty
 } = require("../../util");
-const { httpSend } = require("../../../adapters/network");
+const { httpPOST, httpGET } = require("../../../adapters/network");
 const stats = require("../../../util/stats");
 
-const getHeaderFields = config => {
+const fieldSchema = async config => {
+  const fieldArrNames = [];
+  let fieldArr = [];
+  const fieldSchemaMapping = await httpGET(
+    `https://${config.munchkinId}.mktorest.com/rest/v1/leads/describe2.json`,
+    {
+      params: {
+        access_token: await getAccessToken(config)
+      }
+    }
+  );
+  if (fieldSchemaMapping && fieldSchemaMapping.success) {
+    fieldArr = fieldSchemaMapping.response.data.result[0].fields;
+    fieldArr.forEach(field => {
+      fieldArrNames.push(field.name);
+    });
+    // const fieldNames = fieldArrNames;
+  } else {
+    throw new CustomError("Failed to fetch Marketo Field Schema", 400);
+  }
+  return fieldArrNames;
+};
+
+const getHeaderFields = (config, fieldArrNames) => {
   const { columnFieldsMapping } = config;
+
+  columnFieldsMapping.forEach(colField => {
+    if (fieldArrNames && !fieldArrNames.includes(colField.to)) {
```

Comment:
Here we are checking if columnFieldsMapping is a subset of fieldArrNames. Does the destination accept data only for a partial set of fields?

Reply:
`columnFieldsMapping` stores the fields that are configured by the user on UI. `fieldArrNames` is storing all the fields that are supported by Marketo. So we are checking if the fields given by the user are actually supported by Marketo, hence the subset.